pkg/probe/http/http.go

```diff
@@ -53,7 +53,9 @@ func NewWithTLSConfig(config *tls.Config, followNonLocalRedirects bool) Prober {
 			DisableKeepAlives:  true,
 			Proxy:              http.ProxyURL(nil),
 			DisableCompression: true, // removes Accept-Encoding header
+			DialContext:        probe.ProbeDialer().DialContext,
```

Comment:
Do we need to set `DialTLSContext` too?   We do support HTTPS via the "scheme" field

Reply:
The transport is smart, it uses the dial function to create the TCP connection and start the TLS handshake on top.
I've added a new combination on the tests for https and http

Reply:
only one Dial is needed